src/net/sourceforge/kolmafia/session/EquipmentManager.java

```diff
@@ -2131,13 +2131,21 @@ public static final SpecialOutfit currentOutfit() {
   }
 
   public static final SpecialOutfit currentOutfit(AdventureResult[] equipment) {
+    // Go through any outfit that any worn item belongs to.
     int hash = SpecialOutfit.equipmentHash(equipment);
-    for (SpecialOutfit outfit : EquipmentDatabase.normalOutfits.values()) {
-      if (outfit == null) {
-        continue;
-      }
-      if (outfit.isWearing(equipment, hash)) {
-        return outfit;
+    List<Integer> checkedOutfits = new ArrayList<>();
```

Comment:
may as well be a Set, yeah? Order doesn't matter, `add` and `contains` want to be fast.

Reply:
It's going to be one or two elements the vast majority of the time, so I'd expect a list to be better.